'use strict';
var util = require('util');
var XsODataError = require('./xsODataError');

module.exports = HttpError;

/**
 * Error indicating an XSOData internal problem, e.g. used in try/catch statements to catch runtime errors
 * @param message
 * @param cause
 * @param context
 * @param statusCode
 * @param innerError
 * @constructor
 */
function HttpError(message, context, cause, statusCode,innerError) {
    XsODataError.call(this, message, cause);
    this.statusCode = statusCode;
    this.context = context;
    this.innerError = innerError;
}

util.inherits(HttpError, XsODataError);

